modules/src/core/ics02_client/client_consensus.rs

```diff
@@ -23,23 +23,30 @@ pub const TENDERMINT_CONSENSUS_STATE_TYPE_URL: &str =
 
 pub const MOCK_CONSENSUS_STATE_TYPE_URL: &str = "/ibc.mock.ConsensusState";
 
-pub trait ConsensusState: Clone + core::fmt::Debug + Send + Sync {
-    type Error;
-
+pub trait ConsensusState: core::fmt::Debug + Send + Sync + AsAny {
     /// Type of client associated with this consensus state (eg. Tendermint)
     fn client_type(&self) -> ClientType;
 
     /// Commitment root of the consensus state, which is used for key-value pair verification.
     fn root(&self) -> &CommitmentRoot;
 
-    /// Wrap into an `AnyConsensusState`
-    fn wrap_any(self) -> AnyConsensusState;
+    fn encode_vec(&self) -> Result<Vec<u8>, Error>;
```

Comment:
I think we decided against including `encode_vec()` directly in the traits, and instead using a trait object safe version of `Protobuf`, right? We can address this in another PR, perhaps #2412; let's just not forget to do it!